crates/vault-registry/src/lib.rs

```diff
@@ -417,6 +417,21 @@ pub mod pallet {
             Self::_set_liquidation_collateral_threshold(currency_pair, threshold);
             Ok(())
         }
+
+        /// Changes the collateral liquidation threshold for a currency (only executable by the Root account)
+        ///
+        /// # Arguments
+        /// * `version` - the semver version, where the zero-index element is the major version
+        /// * `checksum` - the SHA256 checksum of the client binary
+        #[pallet::weight(<T as Config>::WeightInfo::set_liquidation_collateral_threshold())]
+        #[transactional]
+        pub fn set_client_release(origin: OriginFor<T>, version: [u32; 3], checksum: [u8; 32]) -> DispatchResult {
```

Comment:
```suggestion
        pub fn set_client_release(origin: OriginFor<T>, uri: Vec<u8>, code_hash: T::Hash) -> DispatchResult {
```

As discussed, let's bundle the URI string to remove any ambiguity and then also set the `code_hash` of the runtime to apply in `authorize_upgrade`.